proc/arch.go

```diff
@@ -15,38 +16,34 @@ type AMD64 struct {
 	ptrSize                 int
 	breakInstruction        []byte
 	breakInstructionLen     int
-	curgInstructions        []byte
+	gStructOffset           uint64
 	hardwareBreakpointUsage []bool
 }
 
 func AMD64Arch() *AMD64 {
-	var (
-		curg       []byte
-		breakInstr = []byte{0xCC}
-	)
-
-	switch runtime.GOOS {
-	case "darwin":
-		curg = []byte{
-			0x65, 0x48, 0x8b, 0x0C, 0x25, 0xA0, 0x08, // mov %gs:0x8a0,%rcx
-			0x0, 0x0,
-		}
-	case "linux":
-		curg = []byte{
-			0x64, 0x48, 0x8b, 0x0c, 0x25, 0xf0, 0xff, 0xff, 0xff, // mov %fs:0xfffffffffffffff0,%rcx
-		}
-	}
-	curg = append(curg, breakInstr[0])
+	var breakInstr = []byte{0xCC}
 
 	return &AMD64{
 		ptrSize:                 8,
 		breakInstruction:        breakInstr,
-		breakInstructionLen:     1,
-		curgInstructions:        curg,
+		breakInstructionLen:     len(breakInstr),
 		hardwareBreakpointUsage: make([]bool, 4),
 	}
 }
 
+func (a *AMD64) SetCurGInstructions(ver GoVersion, isextld bool) {
+	switch runtime.GOOS {
+	case "darwin":
+		a.gStructOffset = 0x8a0
+	case "linux":
+		if isextld || ver.After(GoVersion{1, 5, 0}) {
```

Comment:
This branching should happen regardless of OS, if I'm not mistaken. Seems like it should be moved down and turned into something like:

```go
if isextld || ver.After(GoVersion{1, 5, 0}) {
        a.gStructOffset += 8
}
```